### PR TITLE
use dexie to persist items

### PR DESCRIPTION
**Pull request by ashlsun**

### CODE REVIEW

File: src/lib/components/StoragePlace.svelte [hunk 1]
```diff
@@ -26,33 +36,66 @@
...
+            if (items)
+              items.select(i + amount)
+          }}
+          onDelete={() => {
+            if (items)
+              items.delete(item.id)
+          }}
+          onUpdate={(updatedItem: StoredItem) => {
+            if (items)
+              items.update(item.id, updatedItem)
```
Comment (ashlsun) [inline src/lib/components/StoragePlace.svelte]: wondering if u have an idea why typescript throws the error "`items` might be null" here, though this element should only appear if items exists... ?

Comment (Rettend) [reply src/lib/components/StoragePlace.svelte]: no error for me there
image

Comment (ashlsun) [reply src/lib/components/StoragePlace.svelte]: oh without the `if (items)` line i mean - maybe typescript doesnt take into account the {#if items} from svelte?

Comment (Rettend) [reply src/lib/components/StoragePlace.svelte]: oh, right. well this works, using `@const` which just creates a local variable inside the template
```svelte
{#if items}
    {@const store = items}
    <div role="group">
      {#each items.list as item, i (item.id)}
        <Item
          bind:item={items.list[i]}
          isSelected={items.selected === i}
          onSelected={(amount = 0) => {
            store.select(i + amount)
          }}
          onDelete={() => {
            store.delete(item.id)
          }}
          onUpdate={(updatedItem: StoredItem) => {
            store.update(item.id, updatedItem)
          }}
        />
        ```

Comment (Rettend) [reply src/lib/components/StoragePlace.svelte]: weird typescript issue, and this is a temp fix. i still haven't figured out how to best work with stores and a db.

**Pull request merged**